src/App.tsx

```diff
@@ -1,10 +1,61 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Todos } from './components/Todos/Todos';
+import classNames from 'classnames';
+import { wait } from './utils/fetchClient';
+
+type TodosFilter = 'All' | 'Active' | 'Completed';
 
 export const App: React.FC = () => {
+  const [queuedTodos, setQueuedTodos] = useState<Todo[]>([]);
+  const [todosFilter, setTodosFilter] = useState<TodosFilter>('All');
+
+  const [error, setError] = useState('');
+
+  const todosFromServer = useRef<Todo[]>([]);
+  const [completedTodosCount, setCompletedTodosCount] = useState<number>(0);
+
+  const countCompletedTodos = useCallback<() => void>(() => {
+    setCompletedTodosCount(
+      todosFromServer.current.filter(todo => !todo.completed).length,
+    );
+  }, [todosFromServer]);
+
+  const updateTodosFromServer = useCallback(() => {
+    getTodos()
+      .then(todosRecieved => {
+        todosFromServer.current = [...todosRecieved];
+        setQueuedTodos([...todosRecieved]);
+        countCompletedTodos();
+      })
+      .catch(() => {
+        setError('noTodos');
+        wait(3000).then(() => setError(''));
+      });
+  }, [countCompletedTodos]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(updateTodosFromServer, []);
+
+  useEffect(() => {
+    let newTodos: Todo[] = [];
+    const allTodos = [...todosFromServer.current];
+
+    if (todosFilter === 'Active') {
+      newTodos = allTodos.filter(todo => !todo.completed);
+    } else if (todosFilter === 'Completed') {
+      newTodos = allTodos.filter(todo => todo.completed);
+    } else {
+      newTodos = allTodos;
+    }
+
+    setQueuedTodos(newTodos);
+  }, [todosFilter]);
```

Comment:
```suggestion
  const filteredTodos = allTodos.filter(todo => { ...
```
There is no need to use additional state and useeffect here

Reply:
Hello,
I can change useEffect to some callback, no problem, but design of these 3 buttons relies on applied filter, how can i change statuses of each button without filter state?
Thank you, Anastasiia

Reply:
I didn't get it, you don't need to have the additional state for filteredTodos, just save it to some var

if you have some questions about it, feel free to ask them in the chat, you can always ping me there
<img width="631" alt="image" src="https://github.com/user-attachments/assets/b5799cf9-29f5-45db-8bb6-360d43e914e0" />